index.js

```diff
@@ -1082,6 +1084,12 @@
         // activates page tracking
         if (trackRoutes) {
           $rootScope.$on(pageEvent, function () {
+            // Avoid tracking routes not defined as a page or which explicitly deactivate tracking
+            if (readFromRoute && !($route.current && $route.current.templateUrl && !$route.current.dontTrack)) {
```

Comment:
This is fairly dense logic because it has a double-negation in it. It would be nice to spread it out so it is more readable.

```javascript
if (readFromRoute) {
  // Avoid tracking undefined routes, routes without a template, and those explicitly marked as do not track
  if (!$route.current || !$route.current.templateUrl || route.current.doNotTrack) {
    return;
  }
}
```